Objects/listobject.c

```diff
@@ -301,26 +301,27 @@ PyList_Insert(PyObject *op, Py_ssize_t where, PyObject *newitem)
     return ins1((PyListObject *)op, where, newitem);
 }
 
-static int
-app1(PyListObject *self, PyObject *v)
+/* internal, used by _PyList_AppendTakeRef */
+int
+_PyList_AppendTakeRefListResize(PyListObject *self, PyObject *newitem)
```

Comment:
Do we need the extra function? 
Converting `app1` to `_PyList_AppendTakeRef` would seem to be enough.

Reply:
I tried something like that and the results are on the bpo issue, but it seems there is a significant benefit to having the separate as-small-as-possible function that always gets inlined as opposed to a slightly longer version that we let the compiler figure out.

Reply:
Presumably the compiler isn't inlining the function because it isn't hot enough.
`PRECALL_NO_KW_LIST_APPEND` and `LIST_APPEND` are only 0.3% of all instructions executed, so maybe it is best *not* to inline.

Do you have benchmark numbers for the standard suite?

Reply:
https://gist.github.com/sweeneyde/6cbbe1c9d216d117370a809c704b6cfc

Geometric mean: 1.00x faster

Reply:
Also, IMO there is a relative lack of list comprehensions in the pyperformance suite.